anomaly-detection/src/main/java/io/pravega/anomalydetection/event/serialization/PravegaDeserializationSchema.java

```diff
@@ -0,0 +1,38 @@
+/**
+ *
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries.
+ *
+ */
+package io.pravega.anomalydetection.event.serialization;
+
+import io.pravega.client.stream.Serializer;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.streaming.util.serialization.AbstractDeserializationSchema;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+
+/**
+ * A deserialization schema adapter for a Pravega serializer.
+ */
+public class PravegaDeserializationSchema<T extends Serializable> extends AbstractDeserializationSchema<T> {
+    private final Class<T> typeClass;
+    private final Serializer<T> serializer;
+
+    public PravegaDeserializationSchema(Class<T> typeClass, Serializer<T> serializer) {
```

Comment:
Yes but it is straightforward to get the type information by passing the value upright.